orchestrator/orchestrator/src/ethereum_event_watcher.rs

```diff
@@ -41,8 +41,12 @@ pub async fn check_for_events<S: Signer + 'static, CS: CosmosSigner>(
 ) -> Result<U64, GravityError> {
     let prefix = contact.get_prefix();
     let our_cosmos_address = cosmos_key.to_address(&prefix).unwrap();
-    let latest_block = get_block_number_with_retry(eth_client.clone()).await;
-    let latest_block = latest_block - block_delay;
+    let mut latest_block = get_block_number_with_retry(eth_client.clone()).await;
+    if latest_block >= block_delay {
+        latest_block = latest_block - block_delay;
+    } else {
+        latest_block = U64::zero();
+    }
```

Comment:
you can also do: `latest_block.saturating_sub(block_delay)`

Reply:
sure